lib/src/notification/notification.component.ts

```diff
@@ -0,0 +1,177 @@
+    @Input() style?: NotificationStyle;
+    @Input() theme?: NotificationTheme;
+    @Input() title?: string;
+    @Input() toggle: boolean;
+    @Input() actions?: Array<NotificationAction>;
+
+    public notificationClassNames: string;
+    public showNotificationBody: boolean = false;
+    public showNotificationTitle: boolean = false;
+    private timerRef: number = null;
+    private defaultTimeout: number = 5000;
+
+    // helper functions
+
+    /**
+     * Get the style class based on the theme passed through the props
+     * @param {string} style The style passed through the props
+     * @returns {void} The style class
+     */
+    private getStyleClass(): void {
+        let styleClass: string = "style-";
+        if (this.style && ["slide-in", "bar"].indexOf(this.style) !== -1) {
+            styleClass += this.style;
+        } else {
+            styleClass += "slide-in";
+        }
+
+        this.notificationClassNames += " " + styleClass;
+
+        this.showNotificationTitle = this.title && this.style === ("style-slide-in" as NotificationStyle);
+        this.showNotificationBody =
+            styleClass === ("style-slide-in" as NotificationStyle) && this.actions && this.actions.length && this.actions.length < 3;
+    }
+
+    /**
+     * Get the theme class based on the theme passed though the props
+     * @returns {void} The theme class
+     */
+    private getThemeClass(): void {
+        let themeClass: string = "theme-";
+        if (this.theme && ["purple", "primary", "danger", "success", "warning", "inverted"].indexOf(this.theme) !== -1) {
+            themeClass += this.theme;
+        } else {
+            themeClass += "purple";
+        }
+        this.notificationClassNames += " " + themeClass;
+    }
+
+    /**
+     * Get the position class based on the position and style passed through the props
+     * @returns {void} The position class
+     */
+    private getPositionClass(): void {
+        let positionClass: string;
+        const position: string = this.position;
+        if (this.style && ["slide-in", "bar"].indexOf(this.style) !== -1) {
+            switch (this.style) {
+                case "slide-in":
+                    if (position && ["bottom-left", "bottom-right", "top-left", "top-right"].indexOf(position) !== -1) {
+                        positionClass = position;
+                    } else {
+                        positionClass = "bottom-left";
+                    }
+                    break;
+                case "bar":
+                    if (position && ["top", "bottom"].indexOf(position) !== -1) {
+                        positionClass = position;
+                    } else {
+                        positionClass = "top";
+                    }
+                    break;
+            }
+        } else {
+            // Should default back to `slide-in`
+            if (position && ["bottom-left", "bottom-right", "top-left", "top-right"].indexOf(position) !== -1) {
```

Comment:
```suggestion
            if (["bottom-left", "bottom-right", "top-left", "top-right"].some((pos) => pos === position)) {
```

Reply:
Not necessary .

Reply:
I've recently been using array some for these kinds of conditions because that's what it's meant for. `indexOf` is meant to return the index of the item to be used somehow, while `every` and `some` is meant to verify the existence of all or some of the elements. It also returns a boolean which is easier to work within this condition.